core/mondoo-azure-security.mql.yaml

```diff
@@ -988,66 +999,53 @@ queries:
+        2. Navigate to **Key Vaults**.
+        3. Select a key vault and click on **Properties**.
+        4. Verify that **Purge Protection** is enabled, which ensures that deleted vaults and vault objects are recoverable during the retention period.
 
-          ```bash
-          cnspec run azure -c "azure.subscription.keyVault.vaults.all(properties.enablePurgeProtection == "true")" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
-          cnspec run azure -c "azure.subscription.keyVault.vaults.all(properties.enableSoftDelete == "true")" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
-          ```
-
-        Note: It is also possible to use a client secret for app connection instead of a certificate.
-
-        __cnspec shell__
+        **From Azure CLI:**
 
-        To audit Microsoft Azure with `cnspec shell`:
-
-        1. Launch `cnspec shell`:
-
-          ```bash
-          cnspec shell azure --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
-          ```
+        Verify Purge Protection is enabled for each Key Vault:
 
-        2. Run this query:
-
-          ```mql
-          azure.subscription.keyVault.vaults.all(properties.enablePurgeProtection == "true")
-          azure.subscription.keyVault.vaults.all(properties.enableSoftDelete == "true")
-          ```
+        ```bash
+        az keyvault list --query "[].{Name: name, PurgeProtection: properties.enablePurgeProtection}"
+        ```
+        Check the output for each Key Vault, ensuring that PurgeProtection is set to true.
       remediation: |
         ### Microsoft Azure Portal
 
-        To update using Microsoft Azure Portal:
+        1. Log in to the Azure Portal at https://portal.azure.com.
+        2. Navigate to **Key Vaults**.
+        3. Select a key vault and click on **Properties**.
+        4. Scroll to the **Security** section.
+        5. If **Purge Protection** is not enabled, enable it. Note that once enabled, this setting cannot be disabled.
 
-        1. Log into Microsoft Azure Portal at https://portal.azure.com.
-        2. Go to **Key vaults**.
-        3. for each key vault:
-           a. Select **Properties** and ensure **Soft-delete** is enabled.
-           b. Select **Enable Purge Protection**.
-           c. Select **Save**.
+        **From Azure CLI:**
+
+        Enable Purge Protection for a specific Key Vault:
+
+        ```bash
+        az keyvault update --name <VaultName> --resource-group <ResourceGroupName> --set properties.enablePurgeProtection=true
+        ```
   - uid: mondoo-azure-security-ensure-the-kv-is-recoverable-single
     filters: |
       asset.platform == "azure-keyvault-vault"
       asset.kind == "azure-object"
     mql: |
       azure.subscription.keyVault.vault.properties.enablePurgeProtection == true
-      azure.subscription.keyVault.vault.properties.enableSoftDelete == true
```

Comment:
Is there a reason to remove the `enableSoftDelete` part of the check?
At least until February 2025 this seems to be relevant: https://learn.microsoft.com/en-us/azure/key-vault/general/soft-delete-change

Reply:
```
 In February 2025, Microsoft will enable soft-delete protection on all key vaults, and users will no longer be able to opt out of or turn off soft-delete.
 ```

Reply:
It is not needed anymore, Soft-delete is enabled by default!